connector/forwardconnector/forward.go

```diff
@@ -0,0 +1,134 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package forwardconnector // import "go.opentelemetry.io/collector/connector/forwardconnector"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/internal/sharedcomponent"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	typeStr = "forward"
+)
+
+// NewFactory returns a connector.Factory.
+func NewFactory() connector.Factory {
+	return connector.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		connector.WithTracesToTraces(createTracesToTraces, component.StabilityLevelDevelopment),
+		connector.WithMetricsToMetrics(createMetricsToMetrics, component.StabilityLevelDevelopment),
+		connector.WithLogsToLogs(createLogsToLogs, component.StabilityLevelDevelopment),
+	)
+}
+
+// createDefaultConfig creates the default configuration.
+func createDefaultConfig() component.Config {
+	return &struct{}{}
+}
+
+// createTracesToTraces creates a trace receiver based on provided config.
+func createTracesToTraces(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Traces,
+) (connector.Traces, error) {
+	comp := connectors.GetOrAdd(cfg, func() component.Component {
+		return &forwardConnector{}
+	})
+
+	conn := comp.Unwrap().(*forwardConnector)
+	conn.Traces = nextConsumer
+	return conn, nil
+}
+
+// createMetricsToMetrics creates a metrics receiver based on provided config.
+func createMetricsToMetrics(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Metrics,
+) (connector.Metrics, error) {
+	comp := connectors.GetOrAdd(cfg, func() component.Component {
+		return &forwardConnector{}
+	})
+
+	conn := comp.Unwrap().(*forwardConnector)
+	conn.Metrics = nextConsumer
+	return conn, nil
+}
+
+// createLogsToLogs creates a log receiver based on provided config.
+func createLogsToLogs(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Logs,
+) (connector.Logs, error) {
+	comp := connectors.GetOrAdd(cfg, func() component.Component {
+		return &forwardConnector{}
+	})
+
+	conn := comp.Unwrap().(*forwardConnector)
+	conn.Logs = nextConsumer
+	return conn, nil
+}
+
+// This is the map of already created forward connectors for particular configurations.
+// We maintain this map because the Factory is asked trace, metric, and log receivers
+// separately but they must not create separate objects. When the connector is shutdown
+// it should be removed from this map so the same configuration can be recreated successfully.
+var connectors = sharedcomponent.NewSharedComponents()
```

Comment:
I know I am using it as global in other receiver. I think we can use this as member of the Factory.